### DISCUSSION TITLE
Session Hooks

### DISCUSSION THREAD

**Discussion by merlindiavova** [orbitalquark/textadept#422] [Ideas]
Hi,

I'm slowly porting my workflow to Textadept for it to become my main editor.
So far it's been super easy, fantastic!

Part of my workflow is seamlessly switching from project to project. So I created a `projects` module.
I'm trying not introduce new concepts, so I save each project as a separate `*.session` file. 
The module allows me to switch, edit and create new project sessions, with efficient keybindings.

There is one area I'm stuck on. If my module is not used to load a project/session, I have no idea which session file was loaded.

Would it be possible to add the `session_file` to events.SESSION_LOAD and events.SESSION_SAVE?

Thanks

**Comment by orbitalquark** [comment]
That is possible, but before we change the API, could you somehow make 
use of the `session` table argument to save/load your project name or 
some other sort of identifier other than filename?

**Reply by merlindiavova** [reply]
Yes!

I did try this at first, however the custom session data was not being saved. However, I found out the [reason why](https://github.com/orbitalquark/textadept/discussions/340#discussioncomment-6391508).

I have successfully managed to save and load data from sessions. All is good and working as expected.

Thanks for the nudge!
